src/relation/mod.rs

```diff
@@ -260,13 +260,17 @@ impl Map {
         }
     }
 
-    /// Compute the schema and exprs of the reduce
-    fn schema_exprs(named_exprs: Vec<(String, Expr)>, input: &Relation) -> (Schema, Vec<Expr>) {
+    /// Compute the schema and exprs of the map
+    fn schema_exprs(named_exprs: Vec<(String, Expr)>, input: &Relation, filter: &Option<Expr>) -> (Schema, Vec<Expr>) {
+        let input_data_type = match filter {
```

Comment:
This is a typical application case of [if let](https://doc.rust-lang.org/rust-by-example/flow_control/if_let.html).

It would slightly improve readability